### PR TITLE
add the grpc-health-checking to agnhost

### PR DESCRIPTION
**Pull request by SergeyKanzhelev**
/kind feature
/sig node
/sig testing

#### What this PR does / why we need it:

In order to better test GRPC probes (https://github.com/kubernetes/enhancements/issues/2727) and do not require heavy etcd image for e2e, this small addition to agnhost will be very handy.

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

KEP: https://github.com/kubernetes/enhancements/issues/2727

### CODE REVIEW

File: test/images/agnhost/grpc-health-checking/grpc-health-checking.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+var (
+	port              int
+	delayUnhealthySec int
+)
+
+func init() {
+	CmdGrpcHealthChecking.Flags().IntVar(&port, "port", 5000, "Port number.")
+	CmdGrpcHealthChecking.Flags().IntVar(&delayUnhealthySec, "delay-unhealthy-sec", 10, "Number of seconds to delay before start reporting unhealthy.")
```
Comment (aojea) [inline test/images/agnhost/grpc-health-checking/grpc-health-checking.go:51]: time based test use to be flaky, maybe adding some handler that turns it healthy or unhealthy is a better solution to have more control

File: test/images/agnhost/grpc-health-checking/grpc-health-checking.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/health/grpc_health_v1"
+	"google.golang.org/grpc/status"
+)
+
+// CmdGrpcHealthChecking is used by agnhost Cobra.
+var CmdGrpcHealthChecking = &cobra.Command{
+	Use:   "grpc-health-checking",
+	Short: "Starts a simple grpc health checking endpoint",
+	Long:  "Starts a simple grpc health checking endpoint --port to serve on. The endpoint returns SERVING for the first --delay-unhealthy-sec, and NOT_SERVING after this.",
```
Comment (aojea) [inline test/images/agnhost/grpc-health-checking/grpc-health-checking.go:39]: the KEP also mentions 
>   // The service name can be the empty string (i.e. "").                         //+
>  Service string `json:"service" protobuf:"bytes,2,opt,name=service"`  

do we want to add an option to test by service too?

File: test/images/agnhost/grpc-health-checking/grpc-health-checking.go (hunk 1 above)
Comment (SergeyKanzhelev) [reply test/images/agnhost/grpc-health-checking/grpc-health-checking.go:51]: I was following `liveness` example, I agree this is potentially flaky. The nature of probes tests, however, makes it flaky to write non-flaky tests. It's hard to know deterministically when probes start either. So with enough time buffers, this should be ok. 

lmk if you have specific suggestions how to make it better, I'm open for ideas

File: test/images/agnhost/grpc-health-checking/grpc-health-checking.go (hunk 2 above)
Comment (SergeyKanzhelev) [reply test/images/agnhost/grpc-health-checking/grpc-health-checking.go:39]: right, good suggestion. Adding...

File: test/images/agnhost/grpc-health-checking/grpc-health-checking.go (hunk 1 above)
Comment (aojea) [reply test/images/agnhost/grpc-health-checking/grpc-health-checking.go:52]: I'm not very familiar with probe tests but I was thinking in something ala

```
start pod probe:5000 healthy
curl pod:8080/set_probe?unhealthy
```

Comment (SergeyKanzhelev) [reply test/images/agnhost/grpc-health-checking/grpc-health-checking.go]: make sense. Implemented

File: test/images/agnhost/VERSION [hunk 3]
```diff
@@ -1 +1 @@
-2.34
+2.4
```
Comment (aojea) [inline test/images/agnhost/VERSION:1]:
```suggestion
2.35
```

Comment (SergeyKanzhelev) [reply test/images/agnhost/VERSION:1]: applied.

**Comment by aojea** [comment]
/approve
/hold
I think that is safe to approve here, but I'll keep the hold if you want anybody else to look into it
Feel free to unhold

**Comment by SergeyKanzhelev** [comment]
@aojea do you know if the image will be published automatically once merged?

**Comment by aojea** [comment]
> @aojea do you know if the image will be published automatically once merged?

nope, it has to be  promoted 

https://github.com/kubernetes/k8s.io/commit/e66ab6dd87237b35e7d61bda53cc892e31f57133

https://github.com/kubernetes/kubernetes/tree/master/test/images#promoting-images

**Comment by SergeyKanzhelev** [comment]
/unhold

no more takers for review, proceeding

**Pull request merged**